solr/core/src/java/org/apache/solr/request/json/RequestUtil.java

```diff
@@ -214,37 +216,22 @@ public static void processParams(SolrRequestHandler handler, SolrQueryRequest re
           out = "rows";
         } else if (SORT.equals(key)) {
           out = SORT;
+        } else if ("queries".equals(key)) {
+          for (Map.Entry<String, Object> subEntry : ((Map<String, Object>) entry.getValue()).entrySet()) {
```

Comment:
Pls check before cast, throw exception with meaningful message and assert 400 http code in test.

Reply:
Done.